File: typescript/difference-of-squares/difference-of-squares.ts
export class Squares {

  constructor(private readonly count: number) { }

  get sumOfSquares(): number {
    return sum(sequence(1, this.count).map(square))
  }

  get squareOfSum(): number {
    return square(gauss(this.count))
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }
}

const gauss = (x: number): number => (x * (x + 1)) / 2
const square = (x: number): number => Math.pow(x, 2)
const sequence = (from: number, to: number): number[] => Array(to).fill(1).map((_, i) => i + from)
const sum = (xs: number[]): number => xs.reduce((acc, curr) => acc + curr, 0)

File: solutions/zig/difference-of-squares/1/difference_of_squares.zig
const std = @import("std");

pub fn squareOfSum(n: usize) usize {
    return std.math.pow(usize, (n * (n + 1)) / 2, 2);
}

pub fn sumOfSquares(n: usize) usize {
    var sum: usize = 0;
    for (1..n + 1) |i| {
        sum += std.math.pow(usize, i, 2);
    }

    return sum;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
